cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/ParameterTypeRegistry.java

```diff
@@ -21,10 +16,24 @@
 public final class ParameterTypeRegistry {
     // Pattern.compile(...).pattern() is not necessary, but it helps us take advantage of the IntelliJ's regexp validation,
     // which detects unneeded escapes.
-    private static final List<String> INTEGER_REGEXPS = asList(Pattern.compile("-?\\d+").pattern(), Pattern.compile("\\d+").pattern());
-    private static final List<String> FLOAT_REGEXPS = singletonList(Pattern.compile("-?\\d*(?:[.,]\\d+)?").pattern());
-    private static final List<String> WORD_REGEXPS = singletonList(Pattern.compile("[^\\s]+").pattern());
-    private static final List<String> STRING_REGEXPS = singletonList(Pattern.compile("\"([^\"\\\\]*(\\\\.[^\"\\\\]*)*)\"|'([^'\\\\]*(\\\\.[^'\\\\]*)*)'").pattern());
+    private static final List<String> INTEGER_REGEXPS = asList(
+            Pattern.compile("-?\\d+").pattern(),
+            Pattern.compile("\\d+").pattern()
+    );
+    private static final String SIGN = "[-+]?";
+    private static final String MUST_CONTAIN_NUMBER = "(?=.*[0-9].*)";
+    private static final String SCIENTIFIC_NUMBER = "(?:[0-9]+[{expnt}][-+]?[0-9]+)?";
```

Comment:
Is `{expnt}` standard regexp that is easily portable to other languages? If not, I don't think we should use it, as it will lead to inconsistent behaviour.

Reply:
It is a template variable so we can localize the the regular expression. It is substituted out a few lines down for the localized value. It just has the weird spelling so there are no repeat letters which trips up IDEA's regex inspection.

Reply:
Aha! I missed that. In that case - please disregard my other comments. Very cool!